component/otelcol/exporter/exporter.go

```diff
@@ -0,0 +1,165 @@
+	// configuration.
+	Convert() otelconfig.Exporter
+
+	// Extensions returns the set of extensions that the configured component is
+	// allowed to use.
+	Extensions() map[otelconfig.ComponentID]otelcomponent.Extension
+
+	// Exporters returns the set of exporters that are exposed to the configured
+	// component.
+	Exporters() map[otelconfig.DataType]map[otelconfig.ComponentID]otelcomponent.Exporter
+}
+
+// Exporter is a Flow component shim which manages an OpenTelemetry Collector
+// exporter component.
+type Exporter struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	opts     component.Options
+	factory  otelcomponent.ExporterFactory
+	consumer *lazyconsumer.Consumer
+
+	sched *scheduler.Scheduler
+}
+
+var (
+	_ component.Component       = (*Exporter)(nil)
+	_ component.HealthComponent = (*Exporter)(nil)
+)
+
+// New creates a new Flow component which encapsulates an OpenTelemetry
+// Collector exporter. args must hold a value of the argument type registered
+// with the Flow component.
+//
+// The registered component must be registered to export the
+// otelcol.ConsumerExports type, otherwise New will panic.
+func New(opts component.Options, f otelcomponent.ExporterFactory, args Arguments) (*Exporter, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	consumer := lazyconsumer.New(ctx)
+
+	// Immediately set our state with our consumer. The exports will never change
+	// throughout the lifetime of our component.
+	//
+	// This will panic if the wrapping component is not registered to export
+	// otelcol.ConsumerExports.
+	opts.OnStateChange(otelcol.ConsumerExports{Input: consumer})
+
+	e := &Exporter{
+		ctx:    ctx,
+		cancel: cancel,
+
+		opts:     opts,
+		factory:  f,
+		consumer: consumer,
+
+		sched: scheduler.New(opts.Logger),
+	}
+	if err := e.Update(args); err != nil {
+		return nil, err
+	}
+	return e, nil
+}
+
+// Run starts the Exporter component.
+func (e *Exporter) Run(ctx context.Context) error {
+	defer e.cancel()
+	return e.sched.Run(ctx)
+}
+
+// Update implements component.Component. It will convert the Arguments into
+// configuration for OpenTelemetry Collector exporter configuration and manage
+// the underlying OpenTelemetry Collector exporter.
+func (e *Exporter) Update(args component.Arguments) error {
+	eargs := args.(Arguments)
+
+	host := scheduler.NewHost(e.opts.Logger)
+	host.SetExtensions(eargs.Extensions())
+	host.SetExporters(eargs.Exporters())
+
+	settings := otelcomponent.ExporterCreateSettings{
+		TelemetrySettings: otelcomponent.TelemetrySettings{
+			// TODO(rfratto): create an adapter from zap -> go-kit/log
+			Logger: zap.NewNop(),
+
+			// TODO(rfratto): expose tracing and logging statistics.
+			//
+			// We may want to put off tracing until we have native tracing
+			// instrumentation from Flow, but metrics should come sooner since we're
+			// already set up for supporting component-specific metrics.
```

Comment:
Here's an example of what we can do here once we get around to implementing this:
https://github.com/open-telemetry/opentelemetry-go/blob/example/prometheus/v0.32.1/example/prometheus/main.go#L35

It's going to require updating the OpenTeletry Collector dependency, but we're overdue for that anyway.